airflow/models/dagrun.py

```diff
@@ -1051,6 +1059,7 @@ def _expand_mapped_task_if_needed(ti: TI) -> Iterable[TI] | None:
             old_state = schedulable.state
             if not schedulable.are_dependencies_met(session=session, dep_context=dep_context):
                 old_states[schedulable.key] = old_state
+                self.deactivate_scheduling()
```

Comment:
Just curious, shouldn't we only deactivate a dag run scheduling if none of the tis are ready for scheduling?